internal/pkg/service/stream/api/service/sink.go

```diff
@@ -391,6 +391,36 @@ func (s *service) EnableSink(ctx context.Context, d dependencies.SinkRequestScop
 	return s.mapper.NewTaskResponse(t)
 }
 
+func (s *service) UndeleteSink(ctx context.Context, scope dependencies.SinkRequestScope, payload *api.UndeleteSinkPayload) (res *api.Task, err error) {
+	if err := s.sourceMustExist(ctx, scope.SourceKey()); err != nil {
+		return nil, err
+	}
```

Comment:
Does the source must exists if we want to undelete sink? I know it makes sense to have it like that, so we do not recover sink that has no input.

Reply:
it makes sense to me, if the source doesn't exist, we can't restore the sink. 

@jachym-tousek-keboola